restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/ua/UserAgentManager.java

```diff
@@ -158,8 +160,12 @@ private void clean() {
                     if (logger.isInfoEnabled()) {
                         logger.info("Registration: " + result.getAddressOfRecord() + " didn't respond to OPTIONS and will be removed now");
                     }
-                    registrations.removeRegistration(result);
-                    monitoringService.tell(new UserRegistration(result.getUserName(), result.getLocation(), false), self());
+                    // Instead of removing registrations we ping the client one last time to ensure it was not a temporary loss
+                    // of connectivity. We don't need to remove the registration here. It will be handled only if the OPTIONS ping
+                    // times out and the related calls from the client cleaned up as well
+                    ping(result.getLocation());
```

Comment:
I think at this point we can cleanup registration since the ping failed 3 times already at this point.